src/api/GoalsAPI/index.ts

```diff
@@ -171,6 +174,42 @@ export const shareMyGoalAnonymously = async (goal: GoalItem, parent: string) =>
   return res;
 };
 
+export const getGoalHints = async (goal: GoalItem) => {
+  let parentGoalTitle = "root";
+  let parentGoalHint = false;
+  if (goal.parentGoalId !== "root") {
+    parentGoalTitle = (await getGoal(goal.parentGoalId))?.title || "";
+    parentGoalHint = (await getGoalHintFromDB(goal.parentGoalId)) || false;
+  }
+  const requestBody = {
+    method: "getHints",
+    installId: getInstallId(),
+    parentTitle: parentGoalTitle,
+    goal: {
```

Comment:
why are we extracting fields here, can't we pass the full goal object. use object deconstruction if you want to discard private fields like participants.

Reply:
@tijlleenders does the request body need the whole goalItem to be passed?

Reply:
not sure

Reply:
> @tijlleenders does the request body need the whole goalItem to be passed?

No - just there are only a few goal fields expected in the API:  
- title
- duration (optional)